src/kagglehub/models_helpers.py

```diff
@@ -77,15 +77,29 @@ def create_model_if_missing(owner_slug: str, model_slug: str) -> None:
             raise (e)
 
 
-def delete_model(owner_slug: str, model_slug: str) -> None:
+def delete_model(
```

Comment:
A lot of this logic is repetitive through out the code base. Is it better to just create a ModelHandle class that contains all the logic to handle all the different variations?

Reply:
Yes, I can tackle that in a follow up pr :)